index.bs

```diff
@@ -5260,51 +5261,64 @@ network.Initiator = {
     type: "parser" / "script" / "preflight" / "other",
     ? columnNumber: js-uint,
     ? lineNumber: js-uint,
+    ? url: text,
     ? stackTrace: script.StackTrace,
-    ? request: network.Request
 };
 </pre>
 
 The <code>network.Initiatior</code> type represents the source of a network
 request.
 
+Note: the purpose of initiator is to allow finding out which resource
+initiated a particular request.
+
 <div algorithm>
 To <dfn>get the initiator</dfn> given |request|:
 
-1. Let |request id| be |request|'s [=request id=].
-
 1. Let |type| be "<code>other</code>".
 
+1. Let |stack trace|, |url|, |column number|, and |line number| all be null.
+
 1. If |request| is a [=CORS-Preflight Request=], set |type| to
    "<code>preflight</code>".
 
-1. TODO: Get the |type|. It's not quite clear how this ought to work; the CDP
-   data depends on whether the navigation was kicked off by the parser or by
-   script (so e.g. inserting an image from script causes the initiator to be
-   "<code>script</code>"), but that doesn't correspond to anything in Fetch.
+1. Otherwise, if |request|'s [=request/initiator type=] is "<code>script</code>"
+  and |request| is [=isTopLevel=], set |type| to "<code>parser</code>".
+
+1. Otherwise, if |request|'s [=request/initiator type=] is "<code>script</code>"
+  and |request| is not [=isTopLevel=], set |type| to "<code>script</code>".
+
+1. Otherwise, if |request|'s [=request/initiator type=] is "<code>css</code>",
```

Comment:
It's not clear to me if this is correct given e.g. `@import` rules (I don't know how CDP behaves)